ADR/0023-git-references-to-furnish-integration-test-scenarios.md

```diff
@@ -0,0 +1,72 @@
+IntegrationTestScenario [CR] as a reference for the integration tests, in order to run the Tekton PipelineRuns.
+
+Below is the existing format to supply the Integration test references:
+
+![](../diagrams/ADR-0023/tekton-bundle-ITS.jpg)
+
+Building and maintaining Tekton bundles is cumbersome for users when providing custom tests.
+
+Resolving tasks from a git repository is more natural for developers and allows for faster iteration. Hence, users
+should be able to reference their Tekton pipeline definitions that are hosted in git repositories ([RHTAP-402]).
+
+To support the functionality of using git references, Tekton resolvers need to be enabled. As part of [PLNSRVCE-1030], 
+pipeline service adds support for the [Tekton-resolvers] including git resolvers which should in turn give us the ability
+to expand the support for resolvers in the IntegrationTestScenarios.
+
+## Decision
+
+Integration service now migrates to the new version of the IntegrationTestScenario CRD having the expanded functionality
+to allow using different types of Tekton resolvers ex: bundles-resolver, cluster-resolver, hub-resolver along with 
+the git-resolver. This gives the extensibility to permit the use of different resolvers in the future.
+
+The new format of adding the Integration Test Scenarios would look like:
+
+
+![](../diagrams/ADR-0023/git-references-ITS.jpg)
+
+Users git repository path to their test code can now provide the information or the location of their IntegrationTestScenarios
+through Git references like **Github URL**, **Git Options** (**Revision** & **Path in repository**). 
+
+The information provided by the users in the above form will then be consumed by the integration service with the help of IntegrationTestScenario CR and will be used to run the integration tests against the Application Snapshot in question.
+
+Example of the tekton pipeline definition with git resolver:
+
+![](../diagrams/ADR-0023/tekton-pipeline-definition-git-resolver.jpg)
+
+
+
+## Consequences
+
+* Better user experience through Tekton pipeline definitions that are hosted in git repositories which are easier to 
+maintain as opposed to build and maintain the Tekton bundle images.
+
+* Resolvers are currently in Technology Preview for the OpenShift Pipelines 1.9. Technology Preview features are not
+supported with Red Hat production service level agreements (SLAs) and might not be functionally complete.
+Red Hat does not recommend using them in production. These features only provide early access to upcoming product features,
+enabling customers to test functionality and provide feedback during the development process.
+
+
+## Footnotes
+
+* The risk mentioned in the consequences has been [accepted] by the PM(s).
+* With the current design, there will be no direct support for the Tekton bundles. 
+  It could only be supported through the tekton bundle resolver.
```

Comment:
Does the ADR need to mention anything about migrating existing scenarios to the new format (STONEINTG-356)?

Reply:
Addressed at line [#48](https://github.com/redhat-appstudio/book/pull/92/files#diff-4adb99d2e0cdfd21098edf4375e38cd0ca48caeb8f285113121a39634a3989abR48)

Reply:
That looks good to me, thank you!